william-wtr92/instamint
# feat(publications): 🚀 create a publication, publications and comment actions

# As a minter, I can see a minter publications list and interact with

- ### [ITM-30](https://project-william.atlassian.net/browse/ITM-30) As a MINTER, I can see on a MINTER's profile, the list of his publications
- ### [ITM-57](https://project-william.atlassian.net/browse/ITM-57) As a MINTER, I can upload an original content
- ### [ITM-31](https://project-william.atlassian.net/browse/ITM-31) As a MINTER, I can like/dislike an NFT
- ### [ITM-32](https://project-william.atlassian.net/browse/ITM-32) As a MINTER, I can comment on an NFT
- ### [ITM-34](https://project-william.atlassian.net/browse/ITM-34) As a MINTER, I can load more comments

This pull request main focus is publications and all the actions around it, such as comment, like and interact with people in it.

## Proposed Changes

Describe the changes you have made. For lists of significant changes, it may be helpful to structure them in list form:

- Added :

  - View : Component to display publications list
  - View : Modal component to display specifically a publication
  - View : Add publication modal with image cropping system
  - View : Responsiveness for all these components and profile page
  - Server : Publication API routes for creation, liking and fetching
  - Server : Publications comments API routes for sending, replying and fetching

- Fixed :
  - View : BaseLayout correct responsiveness display

## Type of Change

What type of change are you making? (Check all that apply)


## How Has This Been Tested?

Explain the tests you conducted to validate your changes. Provide clear instructions so that the code review can
replicate your tests and confirm the results.

To test this ticket, follow these instructions:

- Run at root `pnpm install`
- Do migrations with `pnpm run kn:latest`
- Run the api business, api files and the front end, then test the functionality on the following paths:
  `http://localhost:3000/profile/[username]`
- Create a publication and then play with it, you can send a comment, reply to your comments or delete the ones you want to remove. You can like/unlike your publications too

## Checklist

Before submitting your PR, please check the appropriate boxes to confirm that you have taken the following actions:


## Screens

https://github.com/william-wtr92/instamint/assets/90528753/fc590e17-2aea-4ea3-a5af-2f2e406f2374


[ITM-30]: https://project-william.atlassian.net/browse/ITM-30?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITM-57]: https://project-william.atlassian.net/browse/ITM-57?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITM-31]: https://project-william.atlassian.net/browse/ITM-31?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITM-32]: https://project-william.atlassian.net/browse/ITM-32?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITM-34]: https://project-william.atlassian.net/browse/ITM-34?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
